### DISCUSSION TITLE
[back] api response HttpStatus 관련

### DISCUSSION THREAD

**Discussion by kyu-baek** [3DPong/transcendence#48] [General]
#### 1. nest 에서 권장하는 방식
```
JS 객체일 경우 자동으로 JSON으로 serialize해줌
JS 원시 타입(string, number, boolean)일 경우는 그냥 값을 응답
기본적으로 응답의 status code는 200이고 POST일 경우 201
```

궁금한 건 201 이 ->   CREATED  로 정의 되어 있는데
모든 post 요청이 다 create 를 하는 로직이 아닌데도 그런식으로 201을 res status 로 설정해도 괜찮은가 하는 의문입니다. 

#### 2. Library-specific
```
@Res()로 라이브러리에 종속적인 응답 객체를 주입 받아서 사용
Express 플랫폼을 사용한다면 res.status(200).send()와 같이 응답을 생성할 수 있음
@Res(), @Next()와 같은 라이브러리 종속적인 기능을 사용하면 NestJS는 자동으로 이걸 인식하고 해당 핸들러에 대해서만
Standard 처리 방식을 비활성화함 (둘 다 같이 사용하고 싶으면 @Res({ passthrough: true })와 같이 옵션을 설정해줘야 함)

```

**Comment by JuneParkCode** [comment]
CRUD 에서 C 에 해당하는 경우 POST를 사용하긴 하죠.  UPDATE의 경우라면 PUT으로 처리하고 200을 받아내는건 어떨까요?
혹시 정확히 어느 부분에서 POST를 하는데 CREATED 가 아닌 상황이 나오는건가요?

**Reply by kyu-baek** [reply]
update 로직이었습니다. create 제외하고는 put 처리를 하겠습니다. 감사합니다.
